File: environment-spring-boot/casadocodigo/src/main/java/org/ecorp/casadocodigo/repositories/PaisRepository.java
package org.ecorp.casadocodigo.repositories;

import org.ecorp.casadocodigo.model.Pais;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface PaisRepository extends JpaRepository<Pais, Long> {

}

File: src/main/kotlin/com/tiozao/cdd/loja/repository/PaisRepository.kt
package com.tiozao.cdd.loja.repository

import com.tiozao.cdd.loja.repository.entity.PaisEntity
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface PaisRepository : CrudRepository<PaisEntity, Int> {
    fun findByNome(paisNome: String): Optional<PaisEntity>

}